### PR TITLE
Feature/v1.0.0 first implementation

### PR DESCRIPTION
**Pull request by sviluppo-ipzs**

### CODE REVIEW

File: src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		if (!Files.exists(keyPath)) {
+			Files.createDirectory(keyPath);
+			reload = true;
+		} else {
+			log.debug("{} path exists", keyDirectoryPath);
+		}
+
+		if (!new File(keyFilePath).exists()) {
+			KeyPairGenerator gen = KeyPairGenerator.getInstance("RSA");
+			gen.initialize(2048);
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java]: why RSA and 2048 are constants?

we use EC in the wallet federation

File: src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+			KeyPair keyPair = gen.generateKeyPair();
+
+			Calendar cal = Calendar.getInstance();
+			cal.add(Calendar.YEAR, 1);
+			Date validityEndDate = cal.getTime();
+
+			JWK jwk = new RSAKey.Builder((RSAPublicKey) keyPair.getPublic())
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java]: I should not use only RSA, EC is required anyway.

File: src/main/java/it/ipzs/fedauthority/controller/OnboardEntityController.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	@Autowired
+	JwtUtil jwtUtil;
+	
+	@Autowired
+	private WebClient webclient;
+
+
+	@PostMapping()
+	public ResponseEntity<OnboardEntity> save(@RequestBody OnboardEntity onboardEntity) throws ParseException, JOSEException {
+		String wellknownURL = onboardEntity.getUrl() + "/" + OIDCConstants.OIDC_FEDERATION_WELLKNOWN_URL;
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/controller/OnboardEntityController.java]: what if the entity_id ends with a trailing `/`?

please generalize this approach avoiding static symbols in the middle like `/`

File: src/main/java/it/ipzs/fedauthority/oidclib/ClaimOptions.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+		return false;
+	}
+
+	public boolean isEmpty() {
+		return itemsMap.isEmpty();
+	}
+
+	public JSONObject toJSON() {
+		JSONObject idToken = new JSONObject();
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/oidclib/ClaimOptions.java]: id token is something out of scope of federation.

claims and id token must be taken out.

File: src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import it.ipzs.fedauthority.oidclib.exception.OIDCException;
+import it.ipzs.fedauthority.oidclib.util.ArrayUtil;
+import it.ipzs.fedauthority.oidclib.util.Validator;
+
+
+public class GlobalOptions<T extends GlobalOptions<T>> {
+
+	public static final int DEFAULT_EXPIRING_MINUTES = 30;
+
+	public static final String DEFAULT_SIGNING_ALG = "RS256";
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java]: we should take these parameters from a ini file or from the ENV, not statically compiled in the code.

File: src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+	public static final String[] SUPPORTED_ENCRYPTION_ENCODINGS = new String[] {
+		"A128CBC-HS256", "A192CBC-HS384", "A256CBC-HS512", "A128GCM", "A192GCM",
+		"A256GCM"};
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java]: in federation we don't need encryption. What's the purpose of this?

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java [hunk 7]
```diff
@@ -0,0 +1,448 @@
...
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import it.ipzs.fedauthority.oidclib.schemas.OIDCProfile;
+
+@Configuration
+@ConfigurationProperties(prefix = "oidc")
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: openid federation is not oidc anymore, please align this

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java [hunk 8]
```diff
@@ -0,0 +1,448 @@
...
+	}
+
+	public JSONObject toJSON() {
+		JSONObject json = new JSONObject();
+
+		json.put("defaultTrustAnchor", defaultTrustAnchor);
+		json.put("trustAnchors", trustAnchors);
+		json.put("relyingParty", relyingParty.toJSON());
+		json.put("spidProviders", new JSONArray(spidProviders));
+		json.put("cieProviders", new JSONArray(cieProviders));
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: RP, OP, AS, RS, wallet specific roles, should be taken out from this implementation which scope is the openid federation authority

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java [hunk 9]
```diff
@@ -0,0 +1,448 @@
...
+		public JSONObject toJSON() {
+			return new JSONObject()
+					.put("trust-anchor", trustAnchor)
+					.put("cie-provider", cieProvider)
+					.put("relying-party", relyingParty);
+		}
+
+		private String trustAnchor = "127.0.0.1";
+		private String cieProvider = "127.0.0.1";
+		private String relyingParty = "127.0.0.1";
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: why using ipv4 private/local ips as constant in the code?

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java [hunk 10]
```diff
@@ -0,0 +1,448 @@
...
+		public String getJwkFilePath() {
+			return jwkFilePath;
+		}
+
+		public void setJwkFilePath(String jwkFilePath) {
+			this.jwkFilePath = jwkFilePath;
+		}
+
+		public String getMdocJwkFilePath() {
+			return mdocJwkFilePath;
```
Comment (peppelinux) [inline src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: why we have mdoc here?

this is out of the scope of openid federation

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java (hunk 5 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java]: These properties were configured from a previous version of a library shared in the last projects (i.e. eudi-wallet-it-openid-credential-issuer), there will be a refactor about it in a next version.

File: src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java (hunk 6 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java]: These properties were configured from a previous version of a library shared in the last projects (i.e. eudi-wallet-it-openid-credential-issuer), there will be a refactor about it in a next version.

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java (hunk 7 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: Modified, but these properties were configured from a previous version of a library shared in the last projects (i.e. eudi-wallet-it-openid-credential-issuer), there will be a refactor about it in a next version.

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java (hunk 8 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: Modified, but these properties were configured from a previous version of a library shared in the last projects (i.e. eudi-wallet-it-openid-credential-issuer), there will be a refactor about it in a next version.

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java (hunk 9 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: Those are a default values, real params are configured via application.yml.

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java (hunk 10 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: Removed.

File: src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java (hunk 2 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java]: These properties were configured from a previous version of a library shared in the last projects (i.e. eudi-wallet-it-openid-credential-issuer), there will be a refactor about it in a next version.

File: src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java (hunk 1 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java]: These properties were configured from a previous version of a library shared in the last projects (i.e. eudi-wallet-it-openid-credential-issuer), there will be a refactor about it in a next version.

Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java]: Updated

File: src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java (hunk 2 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/config/KeyStoreConfig.java]: Updated

File: src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java (hunk 5 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java]: Updated

File: src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java (hunk 6 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/GlobalOptions.java]: Updated

File: src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java (hunk 7 above)
Comment (sviluppo-ipzs) [reply src/main/java/it/ipzs/fedauthority/oidclib/OidcConfig.java]: Updated

**Pull request merged**